tests/index.ts

```diff
@@ -272,3 +272,44 @@ test('should support the integrity field if present', () => {
     // We should not have made any change to the order of outputted lines (@yarnpkg/lockfile 1.0.0 had this bug)
     expect(yarn_lock).toBe(deduped);
 });
+
+test('prioritizes direct requirements if present', () => {
+    const yarn_lock = outdent`
+    a-package@*:
```

Comment:
Ideally (and I acknowledge this is slightly out of the scope of the PR), we should use a similar `yarn.lock` to test all strategies (i.e. the first two tests in this file). That way it's more clear what's the different behaviour of each strategy.